### DISCUSSION TITLE
How to compile a complete program to obtain usable fcitx5

### DISCUSSION THREAD

**Discussion by rxy-possible** [fcitx/fcitx5#1020] [Q&A] [answered]
Hello teacher, I am a new Chinese programmer who wants to get started with input method development in Debian12 (LXQT desktop environment). Recently, I have been browsing the fcitx5 source code, but I don't know how to compile it, or I'm not sure if the fcitx source code still depends on other source code packages that need to be compiled.I found no error after i do cmake and make.Teacher,could you tell me the entire compile process in this project.I am very grateful.

**Comment by wengxt** [comment, marked as answer]
https://fcitx-im.org/wiki/Compiling_fcitx5

**Reply by rxy-possible** [reply]
Thank you, I am able to successfully compile the project, but when I tried to input Chinese, I encountered an error where Chinese was not available, such as libpunctuation.so not being successfully located as shown in the following figure. Could you please guide me if there are any dynamic libraries that have not been generated or backend services that have not been started?
![screensocantload](https://github.com/fcitx/fcitx5/assets/69881307/16108876-9554-4fcd-8132-e94dcc2507d7)

**Reply by wengxt** [reply]
Likely you're install different package to different directory. In the simplest setup, you should use same -DCMAKE_INSTALL_PREFIX for all package that you want to use. Otherwise, you will need to set FCITX_ADDON_DIRS environment variable to point to all the  directories that points to e.g. prefix1/lib/fcitx:prefix2/lib/fcitx ...
